client/analytics/components/report-table/index.js

```diff
@@ -109,25 +107,17 @@ ReportTable.propTypes = {
 	 * String to display as the title of the table.
 	 */
 	title: PropTypes.string.isRequired,
-	/**
-	 * Name of the property in the primary data totals object which contains the
-	 * total number of items.
-	 */
-	totalsCountField: PropTypes.string,
 };
 
 ReportTable.defaultProps = {
 	getSummary: () => null,
 	tableQuery: {},
-	totalsCountField: '',
 };
 
 export default compose(
 	withSelect( ( select, props ) => {
-		const { endpoint, query, tableQuery } = props;
-		// @TODO allow loading the primary data for the variations table once #926 is fixed.
-		const primaryData =
-			endpoint === 'variations' ? {} : getReportChartData( endpoint, 'primary', query, select );
+		const { endpoint, getSummary, query, tableQuery } = props;
+		const primaryData = getSummary ? getReportChartData( endpoint, 'primary', query, select ) : {};
```

Comment:
`getSummary` will always be truthy because there is a defaultProp set

Reply:
I thought like you but I tested it and it seems `defaultProps` are assigned after `withSelect` is executed. So here, `getSummary` might be false.

Anyway, since that's a behavior that might mislead us in the future, I refactored the code to make it easier to understand: 5c314c75d71c58f505ee6266e2b37add2787d0a0.